packages/snap-preact-components/src/components/Atoms/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,145 @@
+/** @jsx jsx */
+import { h } from 'preact';
+
+import { jsx, css } from '@emotion/react';
+import classnames from 'classnames';
+import { observer } from 'mobx-react-lite';
+import { mergeProps } from '../../../utilities';
+import { ComponentProps, StylingCSS } from '../../../types';
+import { Theme, useTheme, CacheProvider } from '../../../providers';
+import { useState } from 'react';
+
+const CSS = {
+	toggle: ({ activeColor, inactiveColor: deActiveColor, buttonColor, size }: Partial<ToggleProps>) =>
+		css({
+			display: 'flex',
+			alignItems: 'center',
+
+			'&.ss__toggle--disabled': {
+				opacity: '.5',
+				cursor: 'none',
+				pointerEvents: 'none',
+			},
+
+			/* The switch - the box around the slider */
+			'& .ss__toggle__switch': {
+				position: 'relative',
+				display: 'inline-block',
+				width: size,
+				height: `calc(${size} / 2 + 4px)`,
+				margin: '10px',
+			},
+
+			/* Hide default HTML checkbox */
+			'& .ss__toggle__switch input': {
+				opacity: '0',
+				width: '0',
+				height: '0',
+			},
+
+			/* The slider */
+			'& .ss__toggle__slider-box': {
+				position: 'absolute',
+				cursor: 'pointer',
+				top: '0',
+				left: '0',
+				right: '0',
+				bottom: '0',
+				backgroundColor: `${deActiveColor}`,
+				transition: '.4s',
+			},
+
+			'.ss__toggle__slider': {
+				position: 'absolute',
+				content: "''",
+				height: `calc(${size} / 2 - 4px)`,
+				width: `calc(${size} / 2 - 4px)`,
+				left: '4px',
+				bottom: '4px',
+				backgroundColor: `${buttonColor}`,
+				transition: '.4s',
+				zIndex: 1,
+			},
+
+			'& .ss__toggle__switch--filtered .ss__toggle__slider-box': {
+				backgroundColor: `${activeColor}`,
+			},
+
+			'& .ss__toggle__switch--filtered .ss__toggle__slider-box .ss__toggle__slider': {
+				transform: `translateX(calc(${size} / 2 - 4px))`,
+			},
+
+			/* Rounded sliders */
+			'& .ss__toggle__slider-box.ss__toggle__slider-box--round': {
+				borderRadius: '34px',
```

Comment:
This specific pixel radius does not scale properly once the `size` goes above 150px... Perhaps some calculation can be done to get the proper radius here?